compiler/src/iree/compiler/Codegen/TransformStrategies/GPU/MatmulTensorCoreStrategy.h

```diff
@@ -93,12 +104,31 @@ class MatmulStrategy : public AbstractGemmLikeStrategy {
                        /*vectorSize=*/std::nullopt};
   }
 
+  auto map(ArrayRef<AffineExpr> exprs) const {
+    return AffineMap::get(captures.rank(), 0, exprs, ctx);
+  };
+
   // LHS copy is of size mxk.
   MappingInfo lhsCopyMapping() const override {
+    int64_t alignment;
+    SmallVector<int64_t> copySizes;
+    AffineExpr m, n, k;
+    bindDims(ctx, m, n, k);
+    if (captures.lhsIndexing() == map({m, k})) {
+      alignment = this->k();
+      copySizes = SmallVector<int64_t>{blockTileM(), reductionTileSize};
+    } else if (captures.lhsIndexing() == map({k, m})) {
+      alignment = this->m();
+      copySizes = SmallVector<int64_t>{reductionTileSize, blockTileM()};
+    } else {
+      captures.lhsIndexing().dump();
+      llvm_unreachable("unsupported lhs indexing");
+    }
```

Comment:
I'm wondering if we can start generalizing here:
 - define a `getOuterTileSizes()` that gives the blockTile sizes in order followed by reductionTileSize
 - define a `getIterationDims()` that gives iteration dimensions in m,n,k order 
 - do `alignment = getIterationDims()[captures.lhsIndexing().back().cast<AffineDimExpr>().getPosition()]`
 - do `copySizes = map_range(lhsIndexing, [](AffineExpr expr) { getOuterTileSizes()[cast<DimExpr>.getPosition()]; })`

This may be less readable than the current form, but will work for all of lhs/rhs/res right now and will transparently for batched and other contractions.